tests/examples/test_ecephys_data_read.cpp

```diff
@@ -167,37 +168,36 @@ TEST_CASE("ElectricalSeriesReadExample", "[ecephys]")
     REQUIRE(electricalSeriesDataPath == (electricalSeriesPath + "/data"));
     // [example_read_getpath_snippet]
 
-    // [example_read_finish_recording_snippet]
-    // Stop the recording
-    io->stopRecording();
-    io->close();
-    // [example_read_finish_recording_snippet]
-
-    // [example_read_only_snippet]
     // Open an I/O for reading
-    std::shared_ptr<BaseIO> readio = createIO("HDF5", path);
+    // TODO: creating a new I/O makes the read fail.
+    // std::shared_ptr<BaseIO> readio = createIO("HDF5", path);
+    // readio->open();
```

Comment:
@stephprince it looks like there is an issue with how HDF5IO opens an existing file. Read works fine when I use the existing HDF5IO object that was used for write, but when I close the file and then open it again reading any data from the file fails. When I tried this in the  ``testBase.cpp`` it corrupted the file (i.e,. I think it may be opened in "write" mode  instead of "read" or "append").

Reply:
@stephprince this may be related to the issue you encountered on Windows

Reply:
I am having trouble reproducing this on the main branch. 

If I add these lines to the end of `testBase.cpp` I am able to create a new IO object and read the data back in. 

```c++
    io->close();

    // test reopening the file to read the data back in
    std::shared_ptr<BaseIO> readio = createIO("HDF5", path);
    readio->open();
    double* tsBuffer2 = new double[numSamples];
    std::unique_ptr<BaseRecordingData> tsetDset2 =
        readio->getDataSet(dataPath + "/timestamps");
    std::unique_ptr<HDF5::HDF5RecordingData> tsH5Dataset2(
        dynamic_cast<HDF5::HDF5RecordingData*>(tsetDset2.release()));
    readH5DataBlock(tsH5Dataset2->getDataSet(), timestampsType, tsBuffer2);
    std::vector<double> tsRead2(tsBuffer2, tsBuffer2 + numSamples);
    delete[] tsBuffer2;
    REQUIRE(tsRead2 == timestamps);
```

Do you remember if there were any different steps or conditions?

Reply:
Thanks for confirming. A simple test is to just uncomment the two lines here where the `readio` object is being created and then use the `readio` object instead of the original `io` object in the code that follows. We'd also need to call `io.close` first since we don't have a way yet to open as read-only with SWMR-read.

Reply:
I'm not positive since there were some code changes since these comments, but I think the main issue was that `readDataWrapper` was being used here instead of `readElectricalSeriesData` obtained from the `read_io` object.

```c++
auto readElectricalSeriesData = readElectricalSeries->dataLazy();
DataBlock<float> readDataValues = readDataWrapper->values<float>();
```

This test example now works using the `read_io` object, but the file is being re-opened in read/write mode. I will add an option to open in read-only mode and demonstrate that here.

Reply:
> I think the main issue was that `readDataWrapper` was being used here instead of `readElectricalSeriesData`

Makes sense. Thanks for catching that mistake. 

> I will add an option to open in read-only mode and demonstrate that here.

For read-only mode we should also allow SWMR read mode.